spark/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -559,7 +559,9 @@ private[delta] class ConflictChecker(
    *     to handle the row tracking feature being enabled by the winning transaction.
    */
   private def reassignRowCommitVersions(): Unit = {
-    if (!RowTracking.isSupported(currentTransactionInfo.protocol)) {
+    if (!RowTracking.isSupported(currentTransactionInfo.protocol) &&
+      // Type widening relies on default row commit versions to be set.
+      !TypeWidening.isSupported(currentTransactionInfo.protocol)) {
```

Comment:
I don't understand this condition. If row tracking is not supported but type widening is, we will still reassign the default row commit version? How can we have a row commit version without row tracking?

Reply:
We only need a very limited subset of row tracking: that each AddFile action committed when the type widening table feature is supported contains a (correct) defaultRowCommitVersion value.

E.p. we don't require a backfill for older files, that other row tracking information is populated - base row id, high water mark - or that the defaultRowCommitVersion be accessible other than by directly reading it from AddFile action.

I updated the tests for dropping the table feature to run with both row tracking enabled and disabled.